x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/LuceneQueryExpressionEvaluator.java

```diff
@@ -44,19 +45,18 @@ public record ShardConfig(Query query, IndexSearcher searcher) {}
 
     private final BlockFactory blockFactory;
     private final ShardConfig[] shards;
-    private final int docChannel;
 
     private ShardState[] perShardState = EMPTY_SHARD_STATES;
 
-    public LuceneQueryExpressionEvaluator(BlockFactory blockFactory, ShardConfig[] shards, int docChannel) {
+    public LuceneQueryExpressionEvaluator(BlockFactory blockFactory, ShardConfig[] shards) {
         this.blockFactory = blockFactory;
         this.shards = shards;
-        this.docChannel = docChannel;
     }
 
     @Override
     public Block eval(Page page) {
-        DocVector docs = page.<DocBlock>getBlock(docChannel).asVector();
+        // Lucene based operators retrieve DocVectors as first block
+        DocVector docs = page.<DocBlock>getBlock(0).asVector();
```

Comment:
I am not sure about this TBH, what about a guard / assertion that makes sure that the expected block is a `DocBlock`, to avoid `CCE` in case of implementation changes in Lucene operators.

Reply:
Good call, I added an assertion in 8e7a8354ef36cf2af2ca4f695068b8216d1462a8